pkg/costmodel/cluster_helpers.go

```diff
@@ -246,10 +246,10 @@ func buildGPUCostMap(
 		clusterAndNameToType[keyNon] = nodeType
 
 		// If gpu count is available use it to multiply gpu cost
-		if value, ok := gpuCountMap[key]; ok && value != 0 {
+		if value, ok := gpuCountMap[key]; ok {
 			gpuCostMap[key] = gpuCost * value
 		} else {
-			gpuCostMap[key] = gpuCost
+			gpuCostMap[key] = 0
```

Comment:
Is it possible to be in a situation with custom pricing in which:
1. A node's GPU cost is (intentionally) set to something > 0
2. Our GPU count query doesn't pick up the node's GPU, possibly because the node metadata is set up incorrectly or our tracking of node metadata is incomplete

I think this change would cause the node's GPU cost to be 0 when it shouldn't be.

---

But perhaps that's okay and we should just be setting count correctly! If that's the case, then the code change you've made is perfect.